analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParameterNameMatchesOriginal.cs

```diff
@@ -93,4 +91,53 @@ public partial interface IParameterNamesInPartialMethods
         {
         }
     }
+
+    public interface IGenericInterface<A>
+    {
+        void DoSomething(A value);
+        void DoSomething(A value, int intValue);
+        void DoSomethingElse(A value);
+        void DoSomethingElse(A value, ParameterClass parameterClassValue);
+        void TryOneMoreTime(AnotherParameterClass value);
+    }
+    public class ParameterClass { }
+    public class AnotherParameterClass { }
+    public class Implementation : IGenericInterface<ParameterClass>
+    {
+        public void DoSomething(ParameterClass parameter) { }
+
+        public void DoSomethingElse(ParameterClass completelyAnotherName) { }
+
+        public void DoSomething(ParameterClass value, int myValue) { }             // Noncompliant
+//                                                        ^^^^^^^
+        public void DoSomethingElse(ParameterClass value, ParameterClass val) { }  // Noncompliant
+//                                                                       ^^^
+        public void TryOneMoreTime(AnotherParameterClass anotherParameter) { }     // Noncompliant
+//                                                       ^^^^^^^^^^^^^^^^
+    }
+
+    public abstract class BaseClass<T>
+    {
+        public abstract void SomeMethod(T someParameter);
+
+        public abstract void SomeMethod(T someParameter, int intParam);
+    }
+
+    public class ClassOne : BaseClass<int>
+    {
+        public override void SomeMethod(int renamedParam) { }
+
+        public override void SomeMethod(int someParameter, int renamedParam) {  }  // Noncompliant
+//                                                             ^^^^^^^^^^^^
+    }
+
+    public abstract class AbstractClassWithGenericMethod
+    {
+        abstract public void Foo<T>(T val);
+    }
+
+    public class InheritedClassWithDefinition : AbstractClassWithGenericMethod
+    {
+        public override void Foo<T>(T myNewName) { }
```

Comment:
I would say this is a FN, as  the overridden method is still generic, so I do not see why would the parameter name change here. If we opt to solve it, it can be fixed by checking that the overridden parameter type kind is not generic anymore.